itou/siae_evaluations/models.py

```diff
@@ -487,6 +511,18 @@ def state(self):
             for eval_job_app in self.evaluated_job_applications.all()
             for eval_admin_crit in eval_job_app.evaluated_administrative_criteria.all()
         ):
+            if self.evaluation_is_final:
+                if (
+                    not self.reviewed_at
+                    or not self.final_reviewed_at
+                    and any(
+                        eval_admin_crit.submitted_at and eval_admin_crit.submitted_at > self.reviewed_at
+                        for eval_job_app in self.evaluated_job_applications.all()
+                        for eval_admin_crit in eval_job_app.evaluated_administrative_criteria.all()
```

Comment:
cette évaluation que l'on retrouve dans `views.py` ne pourrait-elle pas être renommée `has_unreviewed_criteria` et réutilisée ? ce serait plus clair et éventuellement possible à mettre en cache.

Reply:
Pas de cache, restons simple. Surtout qu’on itère sur des éléments `prefetched`, ça va vite.

C’est pas exactement `has_unreviewed_criteria`, puisque `submitted_at is None` indique aussi `has_unreviewed_criteria`, mais l’evaluation retournerait `False` dans ce cas. Ce serait plus `has_resubmitted_proofs`, qui demande d’avoir vérifié que `reviewed_at` est bien défini. Si on l’extrait, la propriété devrait avoir un gros WARNING de précondition `reviewed_at is not None`.

On passe notre temps à boucler sur les candidatures et leurs justificatifs, rien que regarder la méthode `state` me donne mal au coeur.
- https://github.com/betagouv/itou/blob/1f72729d69f4e3574b572fd605c5d7eefe4ec990/itou/siae_evaluations/models.py#L454-L458
- https://github.com/betagouv/itou/blob/1f72729d69f4e3574b572fd605c5d7eefe4ec990/itou/siae_evaluations/models.py#L462-L466
- https://github.com/betagouv/itou/blob/1f72729d69f4e3574b572fd605c5d7eefe4ec990/itou/siae_evaluations/models.py#L482-L488

Certes, pour cette condition on fait la boucle à deux endroits différents, mais je pense que c’est OK de dupliquer, parce que le code n’est pas vraiment réutilisable.

Reply:
Tres eventuellement on pourrait avoir un `self.any_criteria_matches(lambda c: c.proof_url == "")`, à voir si ça ne fait pas du code un peu smartass, mais plus lisible/compréhensible ?

Reply:
Alors, en fait c’est pas vraiment mutualisable, car la vue part de la `evaluated_job_application`, alors que la property `EvaluatedSiae.state` commence par itérer sur les job applications.
Mais le petit helper pour la lisibilité m’a bien plu : https://github.com/betagouv/itou/pull/1628/commits/37a3745793efd9da49badfacf41ef97c795acb7a.

Reply:
Yes ! Merci pour le helper !